titus-ext/aws/src/main/java/com/netflix/titus/ext/aws/AmazonClientProvider.java

```diff
@@ -29,27 +29,43 @@ public AmazonClientProvider(AwsConfiguration configuration,
     }
 
     public AmazonElasticLoadBalancingAsync getLoadBalancingClient(String accountId) {
-        return loadBalancerClients.computeIfAbsent(accountId, id -> {
-            String region = configuration.getRegion().trim().toLowerCase();
-            AWSCredentialsProvider credentialsProvider = getAwsCredentialsProvider(id);
-            return AmazonElasticLoadBalancingAsyncClientBuilder.standard()
-                    .withCredentials(credentialsProvider)
-                    .withRegion(region)
-                    .build();
-        });
+        AmazonElasticLoadBalancingAsync client = loadBalancerClients.get(accountId);
+        if (client == null) {
+            synchronized (AmazonClientProvider.class) {
```

Comment:
minor: sinchronize on an instance level mutex just to avoid issues when this class is not being used as a singleton (e.g.: tests)

Reply:
Fixed